shell/platform/fuchsia/flutter/accessibility_bridge_unittest.cc

```diff
@@ -0,0 +1,297 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/fuchsia/flutter/accessibility_bridge.h"
+
+#include <gtest/gtest.h>
+#include <lib/async-loop/cpp/loop.h>
+#include <lib/async-loop/default.h>
+#include <lib/fidl/cpp/binding_set.h>
+#include <lib/fidl/cpp/interface_request.h>
+#include <lib/sys/cpp/testing/service_directory_provider.h>
+#include <zircon/types.h>
+
+#include <memory>
+
+#include "flutter/lib/ui/semantics/semantics_node.h"
+#include "flutter/shell/platform/fuchsia/flutter/flutter_runner_fakes.h"
+
+namespace flutter_runner_test {
+
+class AccessibilityBridgeTestDelegate
+    : public flutter_runner::AccessibilityBridge::Delegate {
+ public:
+  void SetSemanticsEnabled(bool enabled) override { enabled_ = enabled; }
+  bool enabled() { return enabled_; }
+
+ private:
+  bool enabled_;
+};
+
+class AccessibilityBridgeTest : public testing::Test {
+ public:
+  AccessibilityBridgeTest()
+      : loop_(&kAsyncLoopConfigAttachToCurrentThread),
+        services_provider_(loop_.dispatcher()) {
+    services_provider_.AddService(
+        semantics_manager_.GetHandler(loop_.dispatcher()),
+        SemanticsManager::Name_);
+  }
+
+  void RunLoopUntilIdle() {
+    loop_.RunUntilIdle();
+    loop_.ResetQuit();
+  }
+
+ protected:
+  void SetUp() override {
+    zx::eventpair a, b;
```

Comment:
Please store these variables as class member and make sure they don't go out of scope.
I have sent an email regarding this.

I am about to land a change which will add a event listener which will delete the semantic tree and close fidl channel with semantics manager when eventpairs go out of scope.

Reply:
Also consider changing their name.

Reply:
Do they have to stay in scope, or the clone we make for a11y, or both?

Reply:
I'll make the change in the test and in the implementation.  I see now what you're saying - the clone is of a different object and may not matter.

Reply:
Implementation was fine - it's held by the platform view.  Updated the test.